herbert_core/utilities/misc/horace_memory.m

```diff
@@ -0,0 +1,77 @@
+function [phys_memory,free_memory]=horace_memory()
+%HORACE_MEMORY Simple system-indepentent utility used by Horace to identify
+% its type for proper auto-configuration
+%
+% Returns:
+% phys_memory  -- physical memory installed on the computer
+% free_memory  -- free memory available for utilites.
+%                 Returns these parameters empty if the memory
+%                 identification does not work.
+
+if ispc
+    [~,sys] = memory();
+    phys_memory = sys.PhysicalMemory.Total;
+    free_memory = sys.PhysicalMemory.Available;
+elseif isunix
+    if ismac
+        [nok,mem_string] = system('top -l 1 -n 0');
+        if ~nok
+            [phys_memory,free_memory] = parse_mac_mem_string(mem_string);
+        end
+    else % normal Unix
+        [nok,mem_string] = system('free | grep Mem');
+        if ~nok
+            [phys_memory,free_memory] = parse_mem_string(mem_string);
+        end
+    end
+    if nok
+        phys_memory = [];
+        free_memory = [];
+    end
+end
```

Comment:
```suggestion
if ispc
    [~,sys] = memory();
    ok = true;
    phys_memory = sys.PhysicalMemory.Total;
    free_memory = sys.PhysicalMemory.Available;
elseif ismac
        [err,mem_string] = system('top -l 1 -n 0');
        ok = err == 0;
        if ok
            [phys_memory,free_memory] = parse_mac_mem_string(mem_string);
        end
elseif isunix
    [err,mem_string] = system('free | grep Mem');
    ok = err == 0;
    if ok
        [phys_memory,free_memory] = parse_mem_string(mem_string);
    end
end

if ~ok
    phys_memory = [];
    free_memory = [];
end
```

Reply:
another "matter of taste" suggestion. I would not fining this more readable then the previous form. May accept it for simplicity, but do not like spending time on such minuscules issues.